panimg/image_builders/dicom.py

```diff
@@ -267,21 +276,52 @@ def _create_itk_from_dcm(self, z_order: int) -> SimpleITK.Image:
         photometric_interpretation = getattr(
             self.ref_header, "PhotometricInterpretation", None
         )
-        if not is_rgb and photometric_interpretation == "MONOCHROME1":
-            if np.issubdtype(dcm_array.dtype, np.floating):
-                dcm_array = -dcm_array
-            else:
-                dcm_array = ~dcm_array
+        self._pixel_values_need_inverting = (
+            not is_rgb and photometric_interpretation == "MONOCHROME1"
+        )
+        dcm_array = self._invert_pixel_values_if_needed(dcm_array)
 
         return SimpleITK.GetImageFromArray(dcm_array, isVector=is_rgb)
 
     def _add_optional_metadata(self, img: SimpleITK.Image):
         for f in OPTIONAL_METADATA_FIELDS:
             value = getattr(self.ref_header, f, "")
             str_value = str(value)
-            if str_value != "":
-                validate_metadata_value(key=f, value=value)
-                img.SetMetaData(f, str_value)
+
+            # Skip empty entries, ITK will not write them anyway
+            if str_value == "":
+                continue
+
+            validate_metadata_value(key=f, value=value)
+            img.SetMetaData(f, str_value)
+
+    def _add_window_level_metadata(self, img: SimpleITK.Image):
```

Comment:
I suggest refactoring this to `_invert_window_level()` and have it together in a single if-statement instead of using an instance member `_pixel_values_need_inverting`. Keeps things a bit more obvious.

Reply:
I don't get entirely what you mean - do you mean putting this step into `_create_itk_from_dcm()` where also the pixel values themselves are inverted?

Reply:
I've changed this such that the inversion happens when adding the metadata to the SITK image object, in the same function where also the other metadata is added. Still not entirely sure whether that's what you meant - but I think it makes sense to do this at that point and not later because the value we get from the DICOM object is a list of values or a single value while the SITK image stores a string representation. So easier to manipulate this value before it's turned into a string.